test/AutoDiff/validation-test/cross_module_differentiation.swift

```diff
@@ -1,7 +1,9 @@
 // RUN: %empty-directory(%t)
-// RUN: %target-build-swift -working-directory %t -parse-as-library -emit-module -module-name cross_module_differentiation_other -emit-module-path %t/cross_module_differentiation_other.swiftmodule -emit-library -static %S/Inputs/cross_module_differentiation_other.swift
-// RUN: %target-build-swift -I%t -L%t %s -o %t/a.out -lcross_module_differentiation_other
-// RUN: %target-run %t/a.out
+// RUN: %target-build-swift-dylib(%t/%target-library-name(cross_module_differentiation_other)) %S/Inputs/cross_module_differentiation_other.swift -emit-module -emit-module-path %t/cross_module_differentiation_other.swiftmodule -module-name cross_module_differentiation_other
+// RUN: %target-build-swift -I%t -L%t %s -o %t/a.out -lcross_module_differentiation_other %target-rpath(%t)
+// RUN: %target-codesign %t/a.out
+// RUN: %target-codesign %t/%target-library-name(cross_module_differentiation_other)
+// RUN: %target-run %t/a.out %t/%target-library-name(cross_module_differentiation_other)
```

Comment:
Are these `RUN` lines maximally simplified?

I wonder what makes the original first line fail work on Windows? It doesn't seem very platform-specific to me:
```
// RUN: %target-build-swift -working-directory %t -parse-as-library -emit-module -module-name cross_module_differentiation_other -emit-module-path %t/cross_module_differentiation_other.swiftmodule -emit-library -static %S/Inputs/cross_module_differentiation_other.swift
```

Reply:
Static linking is not available on Windows, so you must make a shared library.  The code signing is required on macOS.  This really is what it needs to be.